src/components/icons/CryptoIcon.tsx

```diff
@@ -6,85 +6,46 @@ import { ShadowedView } from 'react-native-fast-shadow'
 
 import compromisedIcon from '../../assets/images/compromisedIcon.png'
 import { SPECIAL_CURRENCY_INFO } from '../../constants/WalletAndCurrencyConstants'
-import { useWatch } from '../../hooks/useWatch'
-import { useSelector } from '../../types/reactRedux'
 import { getCurrencyIconUris } from '../../util/CdnUris'
 import { fixSides, mapSides, sidesToMargin } from '../../util/sides'
-import { WalletSyncCircle } from '../progress-indicators/WalletSyncCircle'
 import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
 
 interface Props {
   // Main props - If non is specified, would just render an empty view
-  pluginId?: string // Needed when walletId is not supplied and we still want to get an icon
+  pluginId: string
   tokenId: EdgeTokenId // Needed when it's a token (not the plugin's native currency)
-  walletId?: string // To allow showing the progress ratio sync circle
```

Comment:
So we still want an icon that takes a `walletId`, and automatically figures out stuff like the compromise status. I don't want to shatter this logic into a bunch of different places.

So, I propose creating a `<WalletIcon>` component, which grabs the wallet, looks up the relevant stuff, and then renders a `<CryptoIcon />` with whatever it finds. That way, we have simply split this one component into two, instead of deleting features.